### PR TITLE
#3 - Buildkite nix pipeline added

### PR DESCRIPTION
**Pull request by robcohen**
Tests appear to be passing, github settings need to be changed to check for completed compilation

### CODE REVIEW

**Comment by robcohen** [comment]
Apparently, buildkite check is already activated.

File: release.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+let
+
+  config = {
+    packageOverrides = pkgs: rec {
+      haskellPackages = pkgs.haskell.packages.ghc822.override {
```
Comment (ksaric) [inline release.nix]: Wrong GHC version.

File: release.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        overrides = haskellPackagesNew: haskellPackagesOld: rec {
+          universum           = pkgs.haskell.lib.dontCheck (haskellPackagesNew.callHackage "universum" "1.1.0" {});
+          # LIB
+          cardano-shell  = haskellPackagesNew.callPackage ./cardano-shell.nix { }; 
+          cardano-chain-env = pkgs.haskell.lib.dontCheck (haskellPackagesNew.lib.buildStackProject {
```
Comment (ksaric) [inline release.nix]: We don't need this and I have no clue how this is supposed to work like this.

File: release.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+let
+
+  config = {
+    packageOverrides = pkgs: rec {
+      haskellPackages = pkgs.haskell.packages.ghc822.override {
+        overrides = haskellPackagesNew: haskellPackagesOld: rec {
+          universum           = pkgs.haskell.lib.dontCheck (haskellPackagesNew.callHackage "universum" "1.1.0" {});
+          # LIB
+          cardano-shell  = haskellPackagesNew.callPackage ./cardano-shell.nix { };
```
Comment (ksaric) [inline release.nix]: `=` isn't aligned-

File: scripts/nix/fetchNixpkgs.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{ owner
```
Comment (ksaric) [inline scripts/nix/fetchNixpkgs.nix]: We don't need this file.

File: scripts/nix/nixpkgs-src.json [hunk 5]
```diff
@@ -2,5 +2,6 @@
   "owner": "NixOS",
   "repo": "nixpkgs",
   "rev": "4dd9cd3f69e51dcb69d29a88335f06a544b0e508",
-  "sha256": "0spbpy4c4mvylvdaiic7daw8lfhd4w2mbqx0q2ywmxpqj5hxnhqr"
+  "sha256": "0spbpy4c4mvylvdaiic7daw8lfhd4w2mbqx0q2ywmxpqj5hxnhqr",
```
Comment (ksaric) [inline scripts/nix/nixpkgs-src.json]: We don't need this.

File: stack.yaml [hunk 6]
```diff
@@ -65,5 +65,5 @@ packages:
 # Allow a newer minor version of GHC than the snapshot specifies
 # compiler-check: newer-minor
 nix:
-  shell-file: scripts/nix/stack-shell.nix
+  shell-file: release.nix
```
Comment (ksaric) [inline stack.yaml]: The convention from `cardano-chain` is to use a `scripts` folder and from `cardano-sl` is to use a `pkgs` folder.
I agree that there is no need to pollute the root directory so we can leave it there.

**Comment by ksaric** [comment]
See https://github.com/input-output-hk/cardano-shell/pull/5

File: release.nix (hunk 1 above)
Comment (robcohen) [reply release.nix]: Is there a source of truth for current versioning information that I can refer to in the future? Perhaps something in the cardano-sl repo?

File: release.nix (hunk 2 above)
Comment (robcohen) [reply release.nix]: I believe release.nix is required by Hydra.

File: scripts/nix/fetchNixpkgs.nix (hunk 4 above)
Comment (robcohen) [reply scripts/nix/fetchNixpkgs.nix]: Appears to be used by cardano-sl, not sure how much of their nix configs we should use.

File: scripts/nix/nixpkgs-src.json (hunk 5 above)
Comment (robcohen) [reply scripts/nix/nixpkgs-src.json]: sha256unpacked is needed by fetchNixpkgs, but if we are getting rid of that then it makes sense to not add this.

File: stack.yaml (hunk 6 above)
Comment (robcohen) [reply stack.yaml]: Agree, lets minimize root folder nix scripts. We should be able to get it down to release.nix and shell.nix.

**Pull request closed without merging**